collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -355,42 +388,53 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
     }
 
     private fun quote(text: String) = "\"$text\""
-
-    companion object {
-        private const val DATABASE_VERSION = 2
-    }
 }
 
-private class EntitiesDatabaseMigrator :
-    DatabaseMigrator {
+class EntitiesDatabaseMigrator : DatabaseMigrator {
 
     override fun onCreate(db: SQLiteDatabase) {
-        db.execSQL(
-            """
-            CREATE TABLE IF NOT EXISTS ${ListsTable.TABLE_NAME} (
-                    $_ID integer PRIMARY KEY, 
-                    ${ListsTable.COLUMN_NAME} text NOT NULL,
-                    ${ListsTable.COLUMN_HASH} text
-            );
-            """.trimIndent()
-        )
+        createDbForVersion(db, DATABASE_VERSION)
     }
 
     override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int) {
-        dropAllTablesFromDB(db)
+        if (oldVersion == 1) {
+            throw IllegalStateException("Cannot upgrade from this beta version. Please reinstall Collect!")
+        } else if (oldVersion == 2) {
+            db.addColumn(
+                ListsTable.TABLE_NAME,
+                ListsTable.COLUMN_NEEDS_APPROVAL,
+                "integer",
+                default = "0"
+            )
+        }
     }
-}
 
-private fun dropAllTablesFromDB(db: SQLiteDatabase) {
-    getListsFromDB(db).forEach {
-        db.delete(it)
+    fun createDbForVersion(db: SQLiteDatabase, version: Int) {
+        if (version == 2) {
```

Comment:
We don't need to keep this right if the current version is 3 it is not possible to create db for v2.

Reply:
`onCreate` is where this is actually handled. This just allows any version to be created which I think is easier for testing.

Reply:
> I think is easier for testing.

What exactly would you like to test here?
Let’s say we’re currently using database version 3 and we already have tests for creating the database at that version. Once we upgrade to version 4, we can write new tests for v4 and drop the old ones for v3. So why would we need to keep the v3 tests?

Reply:
We'd still want to maintain tests for upgrading from old versions. We do this in `FormDatabaseMigratorTest` for example - we still want to be able to create "historic" database structures to simulate upgrading from them.